action.yml

```diff
@@ -7,19 +7,37 @@ inputs:
   hax_repository:
     required: true
     default: "github:hacspec/hax"
+  fstar_reference:
+    required: false
+  fstar_repository:
+    required: true
+    default: "github:FStarLang/fstar"
 runs:
   using: "composite"
   steps:
     - uses: DeterminateSystems/nix-installer-action@main
+    - uses: DeterminateSystems/magic-nix-cache-action@main
 
     - name: Install & configure Cachix
       shell: bash
       run: |
         nix-env --quiet -j8 -iA cachix -f https://cachix.org/api/v1/install
         cachix use hax
 
-    - name: Install binaries
+    - name: Install hax
       shell: bash
-      run: nix profile install "$FLAKE"#hax "$FLAKE"#fstar
+      run: nix profile install "$FLAKE"#hax
       env:
         FLAKE: "${{ inputs.hax_repository }}/${{ inputs.hax_reference }}"
+
+    - name: Install F*
+      shell: bash
+      run: |
+        if [[ "$FLAKE" == "github:FStarLang/fstar/" ]]; then
+          REV=$(gh release -R fstarlang/fstar list -L1 --json tagName | jq '.[].tagName' -r)
+          FLAKE="github:FStarLang/fstar/$REV"
+        fi
+        nix profile install "$FLAKE"
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
```
Error: hacspec/hax-actions/fstar-version/action.yml (Line: 42, Col: 19): Unrecognized named-value: 'secrets'. Located at position 1 within expression: secrets.GITHUB_TOKEN
```

Reply:
Ah, that's because it's a reusable workflow, I'm pushing a fix